File: Insertion sort.py
def insertionSortRecursive(arr,n):
	# base case
	if n<=1:
		return
	insertionSortRecursive(arr,n-1)
	
	last = arr[n-1]
	j = n-2
	while (j>=0 and arr[j]>last):
		arr[j+1] = arr[j]
		j = j-1

	arr[j+1]=last
def printArray(arr,n):
	for i in range(n):
		print(arr[i],end=" ")
arr = [12,11,13,5,6]
n = len(arr)
insertionSortRecursive(arr, n)
printArray(arr, n)

File: CPP/insertionSort.cpp
#include <bits/stdc++.h>
using namespace std;
void printArray(int arr[], int n)
{
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    cout << endl;
}
void insertionSort(int arr[], int n)
{
    for (int i = 1; i < n; i++)
    {
        int j = i - 1, key = arr[i];
        while (j >= 0 and key < arr[j])
        {
            arr[j + 1] = arr[j];
            j -= 1;
        }

        arr[j + 1] = key;
        if (i == 5)
        {
            printArray(arr, n);
            break;
        }
    }
}

int main()
{
    int arr[] = {28, 35, 12, 15, 27, 11, 9, 14, 8, 32};
    insertionSort(arr, 5);
    printArray(arr, 5);
}